### PR TITLE
[FISH-7640] Custom Hazelcast System Properties

### PR DESCRIPTION
**Pull request by fturizo**
Documented the system properties implemented in https://github.com/payara/Payara/pull/6318.
These properties have introduced custom behaviour to the Data Grid / Hazelcast and were missing from the current documentation.

This documentation has to be applied retroactively from release `5.58.0`

### CODE REVIEW

File: community/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc [hunk 1]
```diff
@@ -35,4 +35,19 @@ directory | true, false | true
 | fish.payara.node.ssh.timeout | Integer | This option sets the SSH timeout in milliseconds for running commands on SSH nodes. This is a separate timeout to the asadmin or any command specific timeout, this is specifically for the SSH connection used for running certain commands that operate on the remote node itself (e.g. `setup-ssh` and `start-instance`). If set to an invalid value a warning will be logged and the default will be used.  | 1 - Integer.MAX | 120000
 
 | fish.payara.tenantcontrol.blocking.disable | boolean | This option determines if a Payara instance should skip its standard checks for if it is available for a Hazelcast `Tenantable` object. | true, false | true (undefined)
+
+| hazelcast.cp-subsystem.auto-promote | boolean a| When the Payara Server's Data Grid/Hazelcast CP subsystem is enabled (more than 0 members configured), this option will enable or disable an auto-promote behaviour that will allow other server instances to join the Hazelcast CP subsystem when detected by the DAS.
```
Comment (Pandrex247) [inline community/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc:39]: Errant 'a'
```suggestion
| hazelcast.cp-subsystem.auto-promote | boolean | When the Payara Server's Data Grid/Hazelcast CP subsystem is enabled (more than 0 members configured), this option will enable or disable an auto-promote behaviour that will allow other server instances to join the Hazelcast CP subsystem when detected by the DAS.
```

File: community/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc [hunk 2]
```diff
@@ -35,4 +35,19 @@ directory | true, false | true
...
+
+IMPORTANT: The CP subsystem is automatically enabled on the Data Grid/Hazelcast when the `hazelcast.cp-subsystem.cp-member-count` system property is set an integer greater than zero.
+
+| true, false | true
+
+| hazelcast.auto-partition-group | boolean a| Enables/disables Hazelcast group auto-partitioning behaviour in Payara Server's Data Grid.
```
Comment (Pandrex247) [inline community/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc:45]: Errant 'a'
```suggestion
| hazelcast.auto-partition-group | boolean | Enables/disables Hazelcast group auto-partitioning behaviour in Payara Server's Data Grid.
```

File: docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Domain Data Grid And Hazelcast/Configuration.adoc [hunk 3]
```diff
@@ -601,4 +601,16 @@ Using this option to point to a valid Hazelcast configuration file will cause al
...
\ No newline at end of file
+----
+
+== The Hazelcast CP Subsystem
+
+The CP Subsystem is a component of a Hazelcast cluster that builds a strongly consistent layer for a set of distributed data structures. The CP Subsystem withstands server and client failures and offer special features to Payara Server applications.
+
+You can read more about the CP subsystem in the official link:https://docs.hazelcast.com/hazelcast/5.3/cp-subsystem/cp-subsystem[Hazelcast Platform documentation].
+
+To enable the CP subsystem, you will have to configure the `hazelcast.cp-subsystem.cp-member-count` system property to a value greater than zero (read the recommendations in the Hazelcast Platform documentation to pick a suitable member count).
```
Comment (Pandrex247) [inline docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Domain Data Grid And Hazelcast/Configuration.adoc:612]: Is it worth spelling out here in our own docs that this should only be used with member counts greater than 3?

File: enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc [hunk 4]
```diff
@@ -36,4 +36,19 @@ directory | true, false | true
 
 | fish.payara.tenantcontrol.blocking.disable | boolean | This option determines if a Payara instance should skip its standard checks for if it is available for a Hazelcast `Tenantable` object. | true, false | true (undefined)
 
+| hazelcast.cp-subsystem.auto-promote | boolean a| When the Payara Server's Data Grid/Hazelcast CP subsystem is enabled (more than 0 members configured), this option will enable or disable an auto-promote behaviour that will allow other server instances to join the Hazelcast CP subsystem when detected by the DAS.
```
Comment (Pandrex247) [inline enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc:39]: Errant 'a'
```suggestion
| hazelcast.cp-subsystem.auto-promote | boolean | When the Payara Server's Data Grid/Hazelcast CP subsystem is enabled (more than 0 members configured), this option will enable or disable an auto-promote behaviour that will allow other server instances to join the Hazelcast CP subsystem when detected by the DAS.
```

File: enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc [hunk 5]
```diff
@@ -36,4 +36,19 @@ directory | true, false | true
...
+
+IMPORTANT: The CP subsystem is automatically enabled on the Data Grid/Hazelcast when the `hazelcast.cp-subsystem.cp-member-count` system property is set an integer greater than zero.
+
+| true, false | true
+
+| hazelcast.auto-partition-group | boolean a| Enables/disables Hazelcast group auto-partitioning behaviour in Payara Server's Data Grid.
```
Comment (Pandrex247) [inline enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc:45]: Errant 'a'
```suggestion
| hazelcast.auto-partition-group | boolean | Enables/disables Hazelcast group auto-partitioning behaviour in Payara Server's Data Grid.
```

File: community/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc (hunk 2 above)
Comment (fturizo) [reply community/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc:45]: Not an errant `a`, that is an Asciidoc style operator to format the contents of the column it prefixed to full asciidoc, which allows admonitions to be rendered correctly, see https://docs.asciidoctor.org/asciidoc/latest/tables/format-column-content/

File: docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Domain Data Grid And Hazelcast/Configuration.adoc (hunk 3 above)
Comment (fturizo) [reply docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Domain Data Grid And Hazelcast/Configuration.adoc:612]: That's a good idea.

File: enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc (hunk 4 above)
Comment (fturizo) [reply enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc:39]: See my above comment.

File: enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc (hunk 5 above)
Comment (fturizo) [reply enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc:45]: See my above comment.

File: community/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc (hunk 1 above)
Comment (fturizo) [reply community/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Server Configuration And Management/Configuration Options/System Properties.adoc:39]: Not an errant a, that is an Asciidoc style operator to format the contents of the column it prefixed to full asciidoc, which allows admonitions to be rendered correctly, see https://docs.asciidoctor.org/asciidoc/latest/tables/format-column-content/

**Comment by fturizo** [comment]
Thanks for the review, @Pandrex247. I'll merge it right away and back-port it to the corresponding releases.

**Pull request merged**